src/executor.rs

```diff
@@ -41,7 +41,7 @@ pub struct Executor<'a, T = DefaultThreadId> {
     state: Arc<State<T>>,
 
     /// Capture the invariant lifetime.
-    _marker: PhantomData<&'a Cell<Rc<()>>>,
+    _marker: PhantomData<&'a mut &'a Cell<Rc<()>>>,
```

Comment:
```suggestion
    _marker: PhantomData<&'a mut Cell<Rc<()>>>,
```

If my understanding of variance is correct, this should work too, right?

Reply:
If your understanding of variance is correct, this shouldn't work. For `&'a mut T` lifetime `'a` is covariant, but `T` is invariant. See the rustonomicon's variance table:

https://doc.rust-lang.org/stable/nomicon/subtyping.html#variance